src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/EntryPointTableNode.cs

```diff
@@ -0,0 +1,226 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata.Ecma335;
+
+using Internal.NativeFormat;
+using Internal.Runtime;
+using Internal.Text;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class EntryPointTableNode : ObjectNode, ISymbolDefinitionNode
+    {
+        private struct EntryPoint
+        {
+            public static EntryPoint Null = new EntryPoint(-1, -1, -1, 0);
+            
+            public readonly int MethodIndex;
+            public readonly int FixupIndex;
+            public readonly int SignatureIndex;
+            public readonly int MethodHashCode;
+
+            public bool IsNull => (MethodIndex < 0);
+            
+            public EntryPoint(int methodIndex, int fixupIndex, int signatureIndex, int methodHashCode)
+            {
+                MethodIndex = methodIndex;
+                FixupIndex = fixupIndex;
+                SignatureIndex = signatureIndex;
+                MethodHashCode = methodHashCode;
+            }
+        }
+
+        TargetDetails _target;
+        bool _instanceEntryPoints;
```

Comment:
Should this just be two classes instead of switching on a bool? Seems like every method with interesting logic is going to do different things anyway.

Reply:
Hehe, I kept circling around this for quite a while. Some of the logic like fixup unification is identical just like EntryPointWithBlobVertex is derived from EntryPointVertex, on the other hand the two extra fields (signature and method hash code) are superfluous for the [non-instance] method entries. Thanks for helping me resolve this dilemma, will refactor according to your suggestion.

Reply:
Fixed in 5th commit.